quiz.js

```diff
@@ -5,54 +5,84 @@
 var heightField = document.getElementById("height");
 var charField = document.getElementById("character");
 var submitBTN = document.getElementById("growButton");
-var logPlace;
+var printedTree = "";
 
-var tree = {
+
+
+// makes the user input for height and character update tree keys and building the object
+
+
+function getInput(){
+	// console.log("getInput is running")
+	var tree = {
 	height: 0,
 	character: 0
 }
+	tree.height = heightField.value.substring(0,2);//limits the characters read
+	tree.character = charField.value.charAt(0);
+	buildTree(tree);
+
+};
 
-// I need to make the user input for height and character update tree keys
 
-tree.height = heightField;
-tree.character = charField;
 
 
-// create event listeners for clicks and enters
+
+
+// create event listeners for clicks and enters, also displays alert if nothing is entered
 
 heightField.addEventListener("keyup", function(event){
-	if(event.keyCode == 13) {
-			if(heightField.value == "") {
+	if(event.keyCode === 13) {
+			if(!heightField.value || !charField.value) {
 				alert("Both fields must have a value!");
 			}
 			else {
-				buildTree(height, character); 
+				getInput(); 
```

Comment:
Be very careful about leaving trailing whitespace in your code. It will inevitably lead to merge conflicts. You can see whitespace by adding a plugin called Trailing Spaces to your sublime, or you can add this line to your User Preferences and stop them out right. "trim_trailing_white_space_on_save": true.